crates/common/src/identity.rs

```diff
@@ -35,8 +55,17 @@ impl TryFrom<&str> for AuthId {
 }
 
 impl AuthId {
-    pub fn agent(agent: &AgentId) -> Self {
-        Self::JWT(agent.to_owned())
+    pub fn agent(claims: &JwtClaims, json_pointer: &str) -> Result<Self, IdentityError> {
+        let claims = claims.to_owned();
+
+        if let Some(serde_json::Value::String(external_id)) =
+            serde_json::Value::Object(claims.0.clone()).pointer(json_pointer)
+        {
+            Ok(Self::JWT(JwtId::new(external_id, claims)))
+        } else {
+            warn!("Invalid JWT claims: {:#?}", claims.0);
```

Comment:
It may be instead that the JSON pointer is wrong, maybe instead an error about not being able to get the agent ID from the claims?